lib/client/workflow-client/workflowClient.js

```diff
@@ -0,0 +1,367 @@
+ *
+ * In this case, we need access to the workorder, workflow and result objects.
+ *
+ * @param workorderId
+ */
+WorkflowMediatorService.prototype.getWorkorderSummary = function(workorderId) {
+  var self = this;
+
+  return this.readWorkorder(workorderId).then(function(workorder) {
+    return q.all([self.read(workorder.workflowId), self.getResultByWorkorderId(workorderId)]).then(function(allResults) {
+      var workflow = allResults[0];
+      var result = allResults[1];
+
+      return [workorder, workflow, result];
+    });
+  });
+};
+
+/**
+ *
+ * Finding a result based on a workorder ID
+ *
+ * @param {string} workorderId - The ID of the workorder to filter by.
+ */
+WorkflowMediatorService.prototype.getResultByWorkorderId = function getResultByWorkorderId(workorderId) {
+  return this.listResults().then(function(resultsArray) {
+    return _.find(resultsArray || [], function(result) {
+      return result.workorderId === workorderId;
+    }) || null;
+  });
+};
+
+/**
+ *
+ * Creating a new result object for a workorder ID
+ *
+ * @param workorderId
+ */
+WorkflowMediatorService.prototype.createNewResult = function createNewResult(workorderId) {
+  return this.createResult({status: CONSTANTS.STATUS.NEW_DISPLAY, nextStepIndex: 0, workorderId: workorderId, stepResults: {}});
+};
+
+/**
+ *
+ * Listing All Workflows
+ *
+ * @returns {Promise}
+ */
+WorkflowMediatorService.prototype.list = function listWorkflows() {
+  var promise = this.getErrorAndDoneTopicPromises(this.workflowSyncSubscribers, CONSTANTS.TOPICS.LIST);
+
+  this.mediator.publish(this.workflowSyncSubscribers.getTopic(CONSTANTS.TOPICS.LIST));
+
+  return promise;
+};
+
+/**
+ *
+ * Listing all workorders
+ *
+ * @returns {Promise}
+ */
+WorkflowMediatorService.prototype.listWorkorders = function listWorkorders() {
+  var promise = this.getErrorAndDoneTopicPromises(this.workordersTopics, CONSTANTS.TOPICS.LIST);
+
+  this.mediator.publish(this.workordersTopics.getTopic(CONSTANTS.TOPICS.LIST));
+
+  return promise;
+};
+
+/**
+ *
+ * Reading a single workorder
+ *
+ * @param {string} workorderId - The ID of the workorder to read
+ * @returns {Promise}
+ */
+WorkflowMediatorService.prototype.readWorkorder = function readWorkorder(workorderId) {
+  var promise = this.getErrorAndDoneTopicPromises(this.workordersTopics, CONSTANTS.TOPICS.READ, workorderId);
+
+  this.mediator.publish(this.workordersTopics.getTopic(CONSTANTS.TOPICS.READ), {
+    id: workorderId,
+    topicUid: workorderId
+  });
+
+  return promise;
+};
+
+/**
+ *
+ * Reading A single workflow
+ *
+ * @param {string} workflowId
+ * @returns {Promise}
+ */
+WorkflowMediatorService.prototype.read = function readWorkflow(workflowId) {
+  var promise = this.getErrorAndDoneTopicPromises(this.workflowSyncSubscribers, CONSTANTS.TOPICS.READ, workflowId);
+
+  this.mediator.publish(this.workflowSyncSubscribers.getTopic(CONSTANTS.TOPICS.READ), {id: workflowId, topicUid: workflowId});
```

Comment:
Please consider breaking this into two lines. We should try to avoid the need to scroll right to read full line.